Update IntelliJ EAP to 15; version 142.2491.4.

narbsy opened the pull request:
IntelliJ EAP 15 was released recently, so would be nice to get this out there.

https://confluence.jetbrains.com/display/IDEADEV/IDEA+15+EAP

vitorgalvao commented:
Pinging @sebroeder.

sebroeder commented:
@narbsy thanks for the PR. Would you mind working a little more on the `intellij-idea-eap` cask to bring it in line with all the other Jetbrains casks? I put together a list of steps in #879 that are (hopefully) easy to follow. This would be a great help and if you have any questions I am happy to help out.

sebroeder commented on Casks/intellij-idea-eap.rb:
```diff
@@ -1,19 +1,15 @@
+  version '142.2491.4-custom-jdk-bundled'
+  sha256 'a7ce7ec03940b775963cfaf69b2e39bae23a98746691ef9e84b59c8c00d494ff'
 
   url "http://download-cf.jetbrains.com/idea/ideaIU-#{version}.dmg"
```
I would suggest we keep `version '142.2491.4'` and put the `custom-idk-bundled` suffix in the `url`. See #879 for details.

narbsy commented:
@sebroeder  -- thanks for the review; I've changed to using https, done the suggested version change, and added the dangling plist file to `zap`.

Let me know if I've missed anything!

sebroeder commented:
@narbsy very cool, thank you for your improvements. The cask looks good to me, the only thing you missed is the `name` stanza. If you could add the line `name 'IntelliJ IDEA'` between the `url` and `homepage` stanzas, we would be good to go :+1:

sebroeder commented:
@narbsy sorry, found one more issue: the CloudFlare CDN download link doesn't work over https (at least for me). That’s why I always use the direct download link, in this case `http://download.jetbrains.com/idea/ideaIU-#{version}-custom-jdk-bundled.dmg`. Could you please change the download link and test again whether the download works? Thank you.

narbsy commented:
@sebroeder oh weird about the link. I get a cert error from our firewall at work, but then no issues. I'll use the direct link in case anyone else has the same issue.

re: name, I've included EAP ('IntelliJ IDEA EAP') to be more similar to e.g. RubyMine EAP and a bit more clear.

Thanks for the quick turnaround!

sebroeder commented:
@narbsy thanks again. Looks like I wasn't consistent with regards to `EAP` in the `name` stanza myself, e.g. I put it in in the RubyMine cask but left it out in the AppCode EAP. Never mind.

If you want to be even more awesome, you could volunteer to update the `intellij-idea-ce-eap` cask for the Community Edition, too (it is mostly identical to this PR, with another license and some naming/path/url adjustments).  :sunglasses:

narbsy commented:
@sebroeder I'll take a look tonight :)

narbsy commented:
@sebroeder can you approve + merge this? thanks!

narbsy commented:
@vitorgalvao can you review/merge this one as well? thanks!

vitorgalvao commented:
@narbsy As soon as @sebroeder gives an OK, I’ll merge it. He’s been tireless to make all these casks consistent, and I want to make sure that work is not wasted.

sebroeder commented:
Sorry, missed this one. :+1: from me, ready to be merge @vitorgalvao.

narbsy commented:
Thanks guys :)
